sdk/eventhub/Azure.Messaging.EventHubs/stress/src/Infrastructure/EventTracking.cs

```diff
@@ -0,0 +1,246 @@
+    {
+        eventData.Properties.Add(IndexNumberPropertyName, indexNumber);
+        eventData.Properties.Add(PublishTimePropertyName, DateTimeOffset.UtcNow);
+        eventData.Properties.Add(IdPropertyName, Guid.NewGuid().ToString());
+
+        using (SHA256 sha256Hash = SHA256.Create())
+        {
+            eventData.Properties.Add(EventBodyHashPropertyName, sha256Hash.ComputeHash(eventData.EventBody.ToArray()).ToString());
+        }
+
+        if (!string.IsNullOrEmpty(partition))
+        {
+            eventData.Properties.Add(PartitionPropertyName, partition);
+        }
+    }
+
+    /// <summary>
+    ///   Processes the <see cref="EventData"/> instance held in the <see cref="ProcessEventArgs"/> in order to determine
+    ///   if the event has already been seen, if the event was received out of order, or if the body is invalid.
+    /// </summary>
+    ///
+    /// <param name="args">The <see cref="ProcessEventArgs"/> received from the processor client to be used for processing.</param>
+    /// <param name="metrics">The <see cref="Metrics"/> instance used to send information about the processed event to Application Insights.</param>
+    ///
+    public async void ProcessEventAsync(ProcessEventArgs args, Metrics metrics)
+    {
+        var indexNumber = _checkEvent(args.Data, args.Partition.PartitionId, metrics);
+
+        if (indexNumber % 100 == 0)
+        {
+            await args.UpdateCheckpointAsync(args.CancellationToken).ConfigureAwait(false);
+        }
+    }
+
+    /// <summary>
+    ///   Processes the <see cref="EventData"/> instance held in the <see cref="PartitionEvent"/> in order to determine
+    ///   if the event has already been seen, if the event was received out of order, or if the body is invalid.
+    /// </summary>
+    ///
+    /// <param name="partitionEvent">The <see cref="PartitionEvent"/> received from the consumer client to be used for processing.</param>
+    /// <param name="metrics">The <see cref="Metrics"/> instance used to send information about the processed event to Application Insights.</param>
+    ///
+    public void ConsumeEvent(PartitionEvent partitionEvent, Metrics metrics) => _checkEvent(partitionEvent.Data, partitionEvent.Partition.PartitionId, metrics);
+
+    /// <summary>
+    ///   Gets the assigned partitions for the <see cref="PartitionPublisher"/>. This allows each partition publisher instance to publish
+    ///   to distinct and separate partitions. This is necessary so that missing and duplicate events can be detected by the <see cref="Role.Consumer"/>
+    ///   role without needing access to the <see cref="PartitionPublisher"/> instance.
+    /// </summary>
+    ///
+    /// <param name="partitionCount"></param>
+    /// <param name="roleIndex"></param>
+    /// <param name="partitionIds"></param>
+    ///
+    /// <returns>The array of partitions that a given <see cref="PartitionPublisher"/> should send to.</returns>
```

Comment:
```suggestion
    /// <returns>The set of partitions that a given <see cref="PartitionPublisher"/> should send to.</returns>
```

it's not actually an array...